test/jdk/jdk/jfr/event/profiling/TestCPUTimeSampleThrottling.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2024, SAP SE. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.event.profiling;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Comparator;
+
+import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.jfr.EventNames;
+import jdk.test.lib.jfr.Events;
+
+/**
+ * @test
+ * @key jfr
+ * @requires vm.hasJFR & os.family == "linux"
+ * @library /test/lib
+ * @run main/othervm jdk.jfr.event.profiling.TestCPUTimeSampleThrottling
+ */
+public class TestCPUTimeSampleThrottling {
+
+    public static void main(String[] args) throws Exception {
+        testZeroPerSecond();
+        testThrottleSettings();
+        testThrottleSettingsPeriod();
+    }
+
+    private static void testZeroPerSecond() throws Exception {
+        Asserts.assertEquals(0, countEvents(1000, "0/s").count());
+    }
+
+    private static void testThrottleSettings() throws Exception {
+        int count = countEvents(1000,
+            Runtime.getRuntime().availableProcessors() * 2 + "/s").count();
+        Asserts.assertTrue(count > 0 && count < 3,
+            "Expected between 0 and 3 events, got " + count);
+    }
+
+    private static void testThrottleSettingsPeriod() throws Exception {
+        float rate = countEvents(1000, "1ms").rate();
+        Asserts.assertTrue(rate > 950 && rate < 1050, "Expected around 1000 events per second, got " + rate);
```

Comment:
Fails on linux-aarch64-debug builds:

----------System.out:(0/0)----------
----------System.err:(14/1061)----------
java.lang.RuntimeException: Expected around 1000 events per second, got 245.15503: expected true, was false
	at jdk.test.lib.Asserts.fail(Asserts.java:691)
	at jdk.test.lib.Asserts.assertTrue(Asserts.java:543)
	at jdk.jfr.event.profiling.TestCPUTimeEventThrottling.testThrottleSettingsPeriod(TestCPUTimeEventThrottling.java:65)
	at jdk.jfr.event.profiling.TestCPUTimeEventThrottling.main(TestCPUTimeEventThrottling.java:49)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:573)
	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
	at java.base/java.lang.Thread.run(Thread.java:1576)

JavaTest Message: Test threw exception: java.lang.RuntimeException: Expected around 1000 events per second, got 245.15503: expected true, was false
JavaTest Message: shutting down test

Reply:
This is interesting, it shouldn't happen (assuming that the system has at least one CPU that the CPU wasting code can run alone).